frontend/src/component/feature/FeatureView/FeatureVariants/FeatureEnvironmentVariants/PushVariantsButton/PushVariantsButton.tsx

```diff
@@ -0,0 +1,157 @@
+import {
+    Button,
+    Checkbox,
+    Divider,
+    FormControlLabel,
+    Menu,
+    MenuItem,
+    styled,
+} from '@mui/material';
+import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+import { IFeatureEnvironment } from 'interfaces/featureToggle';
+import { useState } from 'react';
+import { useCheckProjectAccess } from 'hooks/useHasAccess';
+
+const StyledMenu = styled(Menu)(({ theme }) => ({
+    '&>div>ul': {
+        display: 'flex',
+        flexDirection: 'column',
+        justifyContent: 'center',
+        padding: theme.spacing(2),
+        '&>li': {
+            padding: theme.spacing(0),
+        },
+    },
+}));
+
+const StyledButton = styled(Button)(({ theme }) => ({
+    marginTop: theme.spacing(1),
+}));
+interface IPushVariantsButtonProps {
+    current: string;
+    environments: IFeatureEnvironment[];
+    permission: string;
+    projectId: string;
+    onSubmit: (selected: string[]) => void;
+}
+
+export const PushVariantsButton = ({
+    current,
+    environments,
+    permission,
+    projectId,
+    onSubmit,
+}: IPushVariantsButtonProps) => {
+    const [pushToAnchorEl, setPushToAnchorEl] = useState<null | HTMLElement>(
+        null
+    );
+    const pushToOpen = Boolean(pushToAnchorEl);
```

Comment:
In terms of code organisation since this is a computation based on a state, maybe it belongs some lines below instead, but it's a tiny nitpick based on preference.